File: app/src/main/java/com/anokmik/tripassistant/base/OnInteractionListener.java
package com.anokmik.tripassistant.base;

import android.content.Intent;
import android.support.v4.app.DialogFragment;
import android.support.v4.app.Fragment;

public interface OnInteractionListener {

    void onLaunchActivity(Intent intent);

    void onReplace(Fragment fragment, String backStackTag);

    void onShowDialog(DialogFragment dialogFragment);

    void onImmediatePopBack(int flags, String backStackTag);

}

File: app/src/main/kotlin/com/anokmik/tripassistant/base/OnInteractionListener.kt
package com.anokmik.tripassistant.base

import android.app.Activity
import android.content.Intent
import android.support.v4.app.DialogFragment
import android.support.v4.app.Fragment

interface OnInteractionListener {

    fun onLaunchActivity(intent: Intent?)

    fun onReplace(fragment: Fragment, backStackTag: String?)

    fun onShowDialog(dialogFragment: DialogFragment)

    fun onImmediatePopBack(flags: Int, backStackTag: String?)

}